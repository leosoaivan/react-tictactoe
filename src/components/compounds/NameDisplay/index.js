import React from 'react';
import styled from 'styled-components';
import PlayerName from '../../atoms/PlayerName';
import PlayerType from '../../../types/playerPropType';

const Root = styled.div`
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 1fr 100px 1fr;
`;

const PlayerOne = styled(PlayerName)`
  grid-column-start: 1;
  grid-column-end: 2;
  justify-self: end;
  align-self: end;
`;

const PlayerTwo = styled(PlayerName)`
  grid-column-start: 3;
  grid-column-end: 4;
  justify-self: start;
  align-self: end;
`;

const Versus = styled.div`
  font-size: 1.5em;
  color: CornflowerBlue;
  grid-column-start: 2;
  grid-column-end: 3;
  justify-self: center;
  align-self: end;
  padding-bottom: 5px;
`;

const NameDisplay = ({ playerOne, playerTwo }) => (
  <Root>
    <PlayerOne
      name={playerOne.name}
      symbol={playerOne.symbol}
    />
    <Versus>
      versus
    </Versus>
    <PlayerTwo
      name={playerTwo.name}
      symbol={playerTwo.symbol}
    />
  </Root>
);

NameDisplay.propTypes = {
  playerOne: PlayerType,
  playerTwo: PlayerType,
};

NameDisplay.defaultProps = {
  playerOne: {},
  playerTwo: {},
};

export default NameDisplay;
