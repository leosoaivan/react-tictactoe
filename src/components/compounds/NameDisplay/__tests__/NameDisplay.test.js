import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { createSerializer } from 'enzyme-to-json';
import NameDisplay from '..';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('NameDisplay', () => {
  it('Renders the NameDisplay component with two player names and one \'versus\'', () => {
    const tree = renderer.create(
      <NameDisplay />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree.children).toHaveLength(3);
  });
});
