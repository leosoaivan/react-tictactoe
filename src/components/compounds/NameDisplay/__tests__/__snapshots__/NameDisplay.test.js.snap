// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameDisplay Renders the NameDisplay component with two player names and one 'versus' 1`] = `
.c2 {
  box-sizing: border-box;
  font-size: 3em;
  font-family: 'Montserrat',sans serif;
  font-weight: 600;
  margin-left: 0.5em;
  margin-right: 0.5em;
  min-width: 0;
  max-width: calc(100% - 0.5em);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: no-wrap;
}

.c0 {
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 1fr 100px 1fr;
}

.c1 {
  grid-column-start: 1;
  grid-column-end: 2;
  justify-self: end;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
}

.c4 {
  grid-column-start: 3;
  grid-column-end: 4;
  justify-self: start;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
}

.c3 {
  font-size: 1.5em;
  color: CornflowerBlue;
  grid-column-start: 2;
  grid-column-end: 3;
  justify-self: center;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  padding-bottom: 5px;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    ()
  </div>
  <div
    className="c3"
  >
    versus
  </div>
  <div
    className="c4 c2"
  >
    ()
  </div>
</div>
`;
